From 7f136e9ecb4c2d52fc946952c64e27c10d250a91 Mon Sep 17 00:00:00 2001
From: "Yaoming.Qin" <B10945@freescale.com>
Date: Fri, 27 May 2011 20:58:07 +0800
Subject: [PATCH 159/635] ENGR00143607 System crash if use uvc to capture a
 picture.

The root cause is that some uvc will configure the output as MJPEG
as the first supported format, so hardcode the UVC will only output
YUYV format.

Signed-off-by: Yaoming Qin<B10945@freescale.com>
---
 libcamera/CameraHal.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index 4ba8d42..8a446af 100755
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -880,7 +880,7 @@ Pic_out:
             }
             if (mPictureEncodeFormat == 0){
                 mPictureEncodeFormat = mEncoderSupportedFormat[0];
-                mCaptureDeviceCfg.fmt = mCaptureSupportedFormat[0];
+                mCaptureDeviceCfg.fmt = mUvcSpecialCaptureFormat; //For uvc now, IPU only can support yuyv.
                 mPPDeviceNeedForPic = true;
                 CAMERA_HAL_LOG_INFO("Need to do the CSC for Jpeg encoder");
                 CAMERA_HAL_LOG_INFO(" Get the captured format is :%c%c%c%c\n",
-- 
1.8.0

